mcs/class/corlib/System.Reflection.Emit/ModuleBuilder.cs

```diff
@@ -1169,32 +1169,49 @@ public override bool IsDefined (Type attributeType, bool inherit)
 
 		public override object[] GetCustomAttributes (bool inherit)
 		{
-			return base.GetCustomAttributes (inherit);
+			return GetCustomAttributes (null, inherit);
 		}
 
 		public override object[] GetCustomAttributes (Type attributeType, bool inherit)
 		{
-			return base.GetCustomAttributes (attributeType, inherit);
+			if (cattrs == null || cattrs.Length == 0)
+				return new object [] {};
+
+			List<object> results = new List<object> ();
+			for (int i=0; i < cattrs.Length; i++) {
+				Type t = cattrs [i].Ctor.GetType ();
+
+				if (attributeType == null || attributeType.IsAssignableFrom (t))
+					results.Add (cattrs [i].Invoke ());
```

Comment:
How is this supposed to behave in case of a TypeBuilder/ConstructorBuilder?

Reply:
I'm not sure I follow. Cattrs[i] is a CustomAttributeBuilder. Invoke () returns the instantiated attribute using the constructor+arguments we would otherwise serialize. Are you asking if cattrs [i].Invoke () is a TypeBuilder/ConstructorBuilder?

Reply:
@alexanderkyte the `CustomAttributeBuilder` could have been created from a `ConstructorBuilder` using something like:
```
   var typeBuilder = moduleBuilder.DefineType (...);
   var constructorBuilder = typeBuilder.DefineConstructor(...);
   var  customAttributeBuilder = new CustomAttributeBuilder (constructorBuilder, ...);

   moduleBuilder.SetCustomAttribute (customAttributeBuilder);
```
So now there's the question of what should happen:
1. If `typeBuilder` is not baked (its `CreateType()` hasn't been called yet)
2. If the `typeBuilder` is baked
3. A variation on the above where `typeBuilder` was defined in some other moduleBuilder - `moduleBuilder2` but we still stick the `customAttributeBuilder` onto `moduleBuilder`.